src/Umbraco.Core/Routing/DefaultMediaUrlProvider.cs

```diff
@@ -10,11 +12,21 @@ public class DefaultMediaUrlProvider : IMediaUrlProvider
 {
     private readonly MediaUrlGeneratorCollection _mediaPathGenerators;
     private readonly UriUtility _uriUtility;
+    private readonly IUrlAssembler _urlAssembler;
 
+    public DefaultMediaUrlProvider(MediaUrlGeneratorCollection mediaPathGenerators, UriUtility uriUtility, IUrlAssembler urlAssembler)
+    {
+        _mediaPathGenerators = mediaPathGenerators ?? throw new ArgumentNullException(nameof(mediaPathGenerators));
+        _uriUtility = uriUtility;
+        _urlAssembler = urlAssembler;
+    }
+
+    [Obsolete("Use the constructor that has the IUrlAssembler instead. Scheduled to be removed in v15")]
     public DefaultMediaUrlProvider(MediaUrlGeneratorCollection mediaPathGenerators, UriUtility uriUtility)
     {
         _mediaPathGenerators = mediaPathGenerators ?? throw new ArgumentNullException(nameof(mediaPathGenerators));
         _uriUtility = uriUtility;
+        _urlAssembler = StaticServiceProvider.Instance.GetRequiredService<IUrlAssembler>();
     }
```

Comment:
We can just use the `this` pattern instead of setting the properties explicitly.

```suggestion
    [Obsolete("Use the constructor that has the IUrlAssembler instead. Scheduled to be removed in v15")]
    public DefaultMediaUrlProvider(MediaUrlGeneratorCollection mediaPathGenerators, UriUtility uriUtility)
        : this(mediaPathGenerators, uriUtility, StaticServiceProvider.Instance.GetRequiredService<IUrlAssembler>())
    {
    }
```